### ISSUE TITLE
Confusion between `maxContentLength` and `maxBodyLength`

### ISSUE THREAD

**Issue by chinesedfan** [axios/axios#2696]
Backgrounds,

- `axios` has a config option called `maxContentLength`, which means the max response content size.
- And `follow-redirects` provides another option called `maxBodyLength`, which means the max request body size.

I don't think they are the same. So why did we connect them in #1286?
(cc @mividtim @RubenVerborgh @emilyemorehouse)

```
if (config.maxContentLength && config.maxContentLength > -1) {
    options.maxBodyLength = config.maxContentLength;
}
```

I guess we need design a new option called `maxBodyLength` exactly and set its default value correctly to avoid cases like https://github.com/axios/axios/issues/1362#issuecomment-368319644.

**Comment by RubenVerborgh** [comment]
> I don't think they are the same.

I confirm that they are not the same.

follow-redirects has a `maxBodyLength`, because it needs to buffer the request body entity in memory for cases where a `POST` or `PUT` request results in a redirect.

**Comment by chinesedfan** [comment]
Fixed by #2781.
